src/coreclr/System.Private.CoreLib/src/System/Exception.CoreCLR.cs

```diff
@@ -147,17 +147,14 @@ internal void RestoreDispatchState(in DispatchState dispatchState)
                 // in the exception object. This will ensure that when this exception is thrown and these
                 // fields are modified, then EDI's references remain intact.
                 //
-                byte[]? stackTraceCopy = (byte[]?)dispatchState.StackTrace?.Clone();
-                object[]? dynamicMethodsCopy = (object[]?)dispatchState.DynamicMethods?.Clone();
 
                 // Watson buckets and remoteStackTraceString fields are captured and restored without any locks. It is possible for them to
                 // get out of sync without violating overall integrity of the system.
                 _watsonBuckets = dispatchState.WatsonBuckets;
                 _ipForWatsonBuckets = dispatchState.IpForWatsonBuckets;
                 _remoteStackTraceString = dispatchState.RemoteStackTrace;
 
-                // The binary stack trace and references to dynamic methods have to be restored under a lock to guarantee integrity of the system.
-                SaveStackTracesFromDeepCopy(this, stackTraceCopy, dynamicMethodsCopy);
+                SaveStackTracesFromDeepCopy(this, dispatchState.StackTrace);
```

Comment:
```suggestion
                _stackTrace = dispatchState.StackTrace;
```
And delete `SaveStackTracesFromDeepCopy` fcall?

Reply:
Done